lib/std/c.zig

```diff
@@ -145,6 +145,7 @@ pub extern "c" fn ioctl(fd: fd_t, request: c_int, ...) c_int;
 pub extern "c" fn uname(buf: *utsname) c_int;
 
 pub extern "c" fn gethostname(name: [*]u8, len: usize) c_int;
+pub extern "c" fn shutdown(socket: fd_t, how: i32) usize;
```

Comment:
for `how`, I would use a `c_int` instead of `i32`. `c_int` could have different size depending the platform.

Reply:
oh, I see @LemonBoy said the opposite before for `lib/std/os/linux.zig` function.

@LemonBoy, using `c_int` ensure portability: C standard (on which `shutdown()` POSIX is based) defines no size of `int` (only a minimum range, and relative size compared to `short` and `long`)

Reply:
and I think also the return value is wrong: `usize` is unsigned, whereas POSIX `shutdown()` could return `-1` on error. I would use `c_int` here too :smile:

Reply:
> oh, I see @LemonBoy said the opposite before for lib/std/os/linux.zig function.

That's the Linux syscall wrapper, we're directly talking with the kernel there and don't give a damn about C/libc, the types there should be Zig's native ones.

The `c.zig` definitions, on the other hand, should use the C types instead to guarantee the ABI compatibility.

> and I think also the return value is wrong: usize is unsigned, whereas POSIX shutdown() could return -1 on error. I would use c_int here too

Good catch, `c_int` is the correct return type.

Reply:
Ok I've modified the one in `c.zig` to:
```c
pub extern "c" fn shutdown(socket: fd_t, how: c_int) c_int;
```